packages/itwin/tree-widget/src/test/trees/models-tree/ModelsTreeComponent.test.tsx

```diff
@@ -39,8 +45,113 @@ describe("<ModelsTreeComponent />", () => {
+    expect(translateSpy).to.be.calledWith("models");
+  });
+
+  describe("queryModels", () => {
+
+    it("sets available models", async () => {
+      const iModel = {
+        models: {
+          queryProps: sinon.fake.returns(new Promise<ModelProps[]>((resolve) => resolve([{
+            id: "testIdFromQueryModels",
+            modeledElement: {
+              id: "id",
+            },
+            classFullName: "className",
+          }]))),
+        } as unknown as IModelConnection.Models,
+      } as unknown as IModelConnection;
+      const spy = sinon.stub().returns(<></>);
+      sinon.stub(modelsTree, "ModelsTree").returns(<></>);
+      sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+      sinon.stub(UiFramework, "getIModelConnection").returns(iModel);
+      render(
+        <ModelsTreeComponent
+          headerButtons={[spy]}
+        />
+      );
+      await waitFor(() => expect(spy).to.be.calledWith(sinon.match((props: ModelsTreeHeaderButtonProps) => (props.models.length === 1 && props.models[0].id === "testIdFromQueryModels"))));
+    });
+
+    it("returns empty array when quryProps throws error", async () => {
+      const spy = sinon.stub().returns(<></>);
+      sinon.stub(modelsTree, "ModelsTree").returns(<></>);
+      sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+      sinon.stub(UiFramework, "getIModelConnection").returns({} as IModelConnection);
+      render(
+        <ModelsTreeComponent
+          headerButtons={[spy]}
+        />
+      );
+      await waitFor(() => expect(spy).to.be.calledWith(sinon.match((props: ModelsTreeHeaderButtonProps) => (props.models.length === 0))));
+    });
+  });
+
   describe("models tree header buttons", () => {
 
+    it("renders default tree header buttons", () => {
+      const treewHeaderSpy = sinon.stub(treeHeader, "TreeHeader").returns(<></>);
+      sinon.stub(modelsTree, "ModelsTree").returns(<></>);
+      sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+      sinon.stub(UiFramework, "getIModelConnection").returns({} as IModelConnection);
+      render(<ModelsTreeComponent />);
+      expect(treewHeaderSpy).to.be.calledWith(sinon.match((props: TreeHeaderProps) => Children.count(props.children) === 5));
+    });
+
+    it("renders user provided tree header buttons", () => {
+      const treewHeaderSpy = sinon.stub(treeHeader, "TreeHeader").returns(<></>);
+      const spy = sinon.stub().returns(<></>);
+      sinon.stub(modelsTree, "ModelsTree").returns(<></>);
+      sinon.stub(IModelApp.viewManager, "selectedView").get(() => viewport);
+      sinon.stub(UiFramework, "getIModelConnection").returns({} as IModelConnection);
+      render(
+        <ModelsTreeComponent
+          headerButtons={[spy]}
+        />
+      );
+      expect(treewHeaderSpy).to.be.calledWith(sinon.match((props: TreeHeaderProps) => Children.count(props.children) === 1 ));
```

Comment:
I think it would be worth to check if your spy was called (button rendered).

Reply:
fixed